crates/bevy_transform/src/transform_propagate_system.rs

```diff
@@ -12,13 +12,14 @@ pub fn transform_propagate_system(
         (Entity, Option<&Children>, &Transform, &mut GlobalTransform),
         Without<Parent>,
     >,
+    dirty_parent_query: Query<Entity, With<DirtyParent>>,
     mut transform_query: Query<(&Transform, &mut GlobalTransform), With<Parent>>,
     changed_transform_query: Query<Entity, Changed<Transform>>,
     children_query: Query<Option<&Children>, (With<Parent>, With<GlobalTransform>)>,
 ) {
     for (entity, children, transform, mut global_transform) in root_query.iter_mut() {
         let mut changed = false;
-        if changed_transform_query.get(entity).is_ok() {
+        if changed_transform_query.get(entity).is_ok() || dirty_parent_query.get(entity).is_ok() {
```

Comment:
Like you suggested on Discord, I think this is going to be cleaner with an `Option<&DirtyParent>` query parameter, rather than creating a new query entirely.

Reply:
I've switched this to use an `Or` and the same `changed_transform_query`, wdyt?